app/controllers/sdes/SdesCallbackController.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sdes
+
+import models.sdes.NotificationType.FileProcessingFailure
+import models.sdes.SdesCallback
+import models.submission.{Pending, RejectedSDES, RejectedSDESVirus}
+import play.api.Logging
+import play.api.i18n.MessagesApi
+import play.api.libs.json.JsValue
+import play.api.mvc.{Action, ControllerComponents}
+import repositories.submission.FileDetailsRepository
+import services.EmailService
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import utils.DateTimeFormatUtil
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SdesCallbackController @Inject() (
+  cc: ControllerComponents,
+  fileDetailsRepository: FileDetailsRepository,
+  emailService: EmailService,
+  implicit override val messagesApi: MessagesApi
+)(implicit val ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  val callback: Action[JsValue] = Action.async(parse.json) { implicit request =>
+    withJsonBody[SdesCallback] { sdesCallback: SdesCallback =>
+      logger.info(s"Received SDES callback: $sdesCallback")
```

Comment:
A request body may contain sensitive info. Not sure if if we should be logging it. Perhaps just the fields we need which we know are not sensitive.